### PR TITLE
feat: Move edit and delete options to toolbar #957

### LINKED ISSUE #957: Move edit and delete options to toolbar for SponsorsFragment
**Is it a ?**

- [ ] Bug Report

- [x] Feature Request
    
- [ ] Chore
    
### Feature Request

**Feature Description**

Edit and delete options from every item in the list should be moved to the toolbar, allowing the user to long press an item to edit or delete it while also supporting multiple deletions.
Edit option should not be present in case of multiple selections.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by mishuvs**
Fixes #957 

Changes: 

1. moves edit and delete options from every item in the list to the toolbar 
2. allows the user to long press an item to edit or delete it
3. supports multiple deletions
4. removes edit option in case of multiple selections
5. re-clicking the last selected item unselects
6. re-clicking the only selected item resets the toolbar

**Gif for the change:**

**Delete Operation:**

videotogif_2018 05 19_04 42 40

**Edit operation:**

videotogif_2018 05 19_04 45 43

### CODE REVIEW

**Comment by mishuvs** [comment]
@Masquerade0097 can you help me out here about the item selection colour?

**Comment by mishuvs** [comment]
@iamareebjamal I am trying to figure out why colour change on selection is not happening. 
Please review if any other changes are also required.

**Comment by iamareebjamal** [comment]
@mishuvs Thanks, I will

File: app/src/main/java/org/fossasia/openevent/app/core/sponsor/list/SponsorsPresenter.java [hunk 1]
```diff
@@ -47,16 +59,17 @@ public void detach() {
         sponsorChangeListener.stopListening();
     }
 
-    public void updateSponsor(Long sponsorId) {
-        getView().openUpdateSponsorFragment(sponsorId);
+    public void updateSponsor() {
+        getView().openUpdateSponsorFragment(selectedSponsors.entrySet().iterator().next().getKey());
```
[the remark this answers is not in the document]
Comment (mishuvs) [reply app/src/main/java/org/fossasia/openevent/app/core/sponsor/list/SponsorsPresenter.java]: Update option is visible when only one item is selected. This gets the key for that only item.

**Pull request merged**